cmd/gaia/app/genesis.go

```diff
@@ -81,30 +85,49 @@ type GaiaGenTx struct {
 	PubKey  string         `json:"pub_key"`
 }
 
-// Generate a gaia genesis transaction with flags
-func GaiaAppGenTx(cdc *wire.Codec, pk crypto.PubKey, genTxConfig config.GenTx) (
-	appGenTx, cliPrint json.RawMessage, validator tmtypes.GenesisValidator, err error) {
+// GaiaAppGenTx generates a Gaia genesis transaction.
+func GaiaAppGenTx(
+	cdc *wire.Codec, pk crypto.PubKey, genTxConfig config.GenTx,
+) (json.RawMessage, json.RawMessage, tmtypes.GenesisValidator, error) {
 	if genTxConfig.Name == "" {
 		return nil, nil, tmtypes.GenesisValidator{}, errors.New("Must specify --name (validator moniker)")
 	}
 
-	var addr sdk.AccAddress
-	var secret string
-	addr, secret, err = server.GenerateSaveCoinKey(genTxConfig.CliRoot, genTxConfig.Name, "1234567890", genTxConfig.Overwrite)
+	buf := client.BufferStdin()
+	prompt := fmt.Sprintf("Password for account '%s' (default %s):", genTxConfig.Name, DefaultKeyPass)
+
+	keyPass, err := client.GetPassword(prompt, buf)
+	if err != nil && keyPass != "" {
+		// An error was returned that either failed to read the password from
+		// STDIN or the given password is not empty but failed to meet minimum
+		// length requirements.
+		return nil, nil, tmtypes.GenesisValidator{}, err
+	}
+
+	if keyPass == "" {
+		keyPass = DefaultKeyPass
+	}
```

Comment:
what's the thinking here around providing the default password here if nothing is provided? Which situations will this come up in? I was thinking we'd just fail if somebody doesn't provide a pass?

Reply:
I'm lazy and don't want to enter anything? If we think absolutely requiring a user-provided password is a better UX exp, then I can remove it @rigelrozanski .

Reply:
I think a better solution would to have a key type with no password. Using a default password only burns computation time.

Reply:
Mhmmm, I'll need to think on that for a bit. What shall we do for now? Just not supply a default?

Reply:
I don't really have a preference rn. I would like just having no passwords as an option later. (I'll spend some time today writing an intended way to do the keybase refactor)

Reply:
Awesome!